tests/test_kb_dataset.py

```diff
@@ -0,0 +1,127 @@
+                        if example[feature_name]:
+                            counter[feature_name] += 1
+                    else:
+                        counter[feature_name] += len(example[feature_name])
+
+            for k, v in counter.items():
+                print(f"{k}: {v}")
+            print()
+
+    def runTest(self):
+        self.print_statistics()
+        self.test_is_bigbio_schema_compatible()
+        self.test_are_ids_globally_unique()
+        self.test_do_all_referenced_ids_exist()
+
+    def test_is_bigbio_schema_compatible(self):
+        for split in self.dataset_bigbio.values():
+            self.assertEqual(split.features, features)
+
+    def test_are_ids_globally_unique(self):
+        for split in self.dataset_bigbio.values():
+            ids_seen = set()
+            for example in split:
+                self._assert_ids_globally_unique(example, ids_seen=ids_seen)
+
+    def test_do_all_referenced_ids_exist(self):
+        for split in self.dataset_bigbio.values():
+            for example in split:
+                referenced_ids = set()
+                existing_ids = set()
+
+                referenced_ids.update(self._get_referenced_ids(example))
+                existing_ids.update(self._get_existing_referable_ids(example))
+
+                for ref_id, ref_type in referenced_ids:
+                    if ref_type == "event_arg":
+                        self.assertTrue(
+                            (ref_id, "entity") in existing_ids
+                            or (ref_id, "event") in existing_ids
+                        )
+                    else:
+                        self.assertIn((ref_id, ref_type), existing_ids)
+
+    def _assert_ids_globally_unique(
+        self, collection, ids_seen: set, ignore_assertion: bool = False
+    ):
+        if isinstance(collection, dict):
+            for k, v in collection.items():
+                if isinstance(v, dict):
+                    self._assert_ids_globally_unique(v, ids_seen)
+                elif isinstance(v, list):
+                    for elem in v:
+                        self._assert_ids_globally_unique(elem, ids_seen)
+                else:
+                    if k == "id":
+                        if not ignore_assertion:
+                            self.assertNotIn(v, ids_seen)
+                        ids_seen.add(v)
+        elif isinstance(collection, list):
+            for elem in collection:
+                self._assert_ids_globally_unique(elem, ids_seen)
+
+    def _get_referenced_ids(self, example):
+        referenced_ids = []
+
+        for event in example["events"]:
+            for argument in event["arguments"]:
+                referenced_ids.append((argument["ref_id"], "event_arg"))
+
+        for coreference in example["coreferences"]:
+            for entity_id in coreference["entity_ids"]:
+                referenced_ids.append((entity_id, "entity"))
+
+        for relation in example["relations"]:
```

Comment:
if we are not requiring all fields, we can still do something like `example.get("relations", [])` here

Reply:
Vastly prefer the `get` call for edge cases; is there a reason we pass empty list instead of None for "invalid keys"? because I would opt strongly for nones unless you need some type behavior

Reply:
`None` is not iterable so if you use `None` you have to write some code for both cases right? by returning an empty list if the component is not there, you can still iterate over it, but nothing will happen b/c its empty. in other words, you can write these for loops for every sub-component in the schema and it wont break if they are not there.